### DISCUSSION TITLE
Easiest way to use sqlite in vscode extension

### DISCUSSION THREAD

**Discussion by cedric05** [microsoft/vscode-discussions#16] [Extension Development QnA] [answered]
I'm working on an [extension ](https://marketplace.visualstudio.com/items?itemName=ShivaPrasanth.dothttp-code). I want to show history of invoked requests. For this purpose I want to use sqlite. 

Where as sqlite doesn't come by default with vscode/nodejs. most of npm packages are dependent/use  [napi](https://nodejs.org/api/n-api.html). which in turn creates platform level dependencies. 

Vscode does have  store (momento) which is key pair not a table/columnar format.  

Is there alternative to sqlite? 

Operations that I plan to use:
1. query list of records by date
2. insert record.

**Comment by digitarald** [comment, marked as answer]
Could https://sql.js.org/ as WASM-based SQLite engine be an option? Still need to store the blob somewhere though.

**Reply by cedric05** [reply]
will try it out. seems like a good option.

**Reply by burkeholland** [reply]
Did you have some luck with this? Can we close this item?

**Reply by cedric05** [reply]
I were not able to test it out.

**Reply by starball5** [reply]
@digitarald, is the https://www.npmjs.com/package/@vscode/sqlite3 package supposed to be for extension developers? Or is it just a package intended for VS Code itself to use under the hood? I just came across [a Stack Overflow question that asked about it](https://stackoverflow.com/q/76838311/11107541).

**Reply by gziz** [reply]
@starball5 did you manage to make it work?

**Reply by digitarald** [reply]
@starball5 its not specifically made for extension usage, but nothing prevents you from use any node.js package.

**Reply by gziz** [reply]
Thanks, Ultimately I was between jsons or sql.js.
I went with the latter, and I haven't regret it!

**Reply by bmingles** [reply]
@gziz Are there any challenges in sql.js with saving / loading the db? Namely, can it get corrupted if multiple vscode instances are open? Would love to hear additional details of how this is used.
